ReactWindows/ReactNative/ReactRootViewExtensions.cs

```diff
@@ -17,8 +18,17 @@ public static void OnCreate(this ReactRootView rootView, ReactNativeHost host)
             SystemNavigationManager.GetForCurrentView().BackRequested += (sender, e) => OnBackRequested(host, sender, e);
             Window.Current.CoreWindow.Dispatcher.AcceleratorKeyActivated += (sender, e) => OnAcceleratorKeyActivated(host, sender, e);
         }
+        public static void OnCreate(this ReactRootView rootView, ReactNativeViewHost host)
```

Comment:
why use extension instead of adding the method to ReactRootView directly?

Reply:
I think the model @rozele uses is to keep just method that use internal state of class X on class X, and everything else to move to extensions.From a user's point of view it looks the same, whereas on the implementation side the difference between "core" and "helpers/convenience" is clearer.
Note: this is just what I inferred from what I've seen.

---
In reply to: [166073907](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166073907) [](ancestors = 166073907)

Reply:
I thought the guidance is to use extensions only when we have no access to the original class and it can't be subclassed but if we have a rule like this is may be good to document.

---
In reply to: [166090891](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166090891) [](ancestors = 166090891,166073907)

Reply:
I don't know if I was judiciously applying any rules here. In this specific case, I didn't like adding the theme resources in the ReactRootView class, and once I had the `OnCreate` helper pulled into an extension method, it made sense for the others related to RootView lifecycle to follow suit.

In general, yes, I agree with what @reseul said that is nice to keep a class lean. If you can implement something entirely with public APIs, why not put it in an extension method? Static methods have a smaller memory footprint than instance methods. It also forces contributors to think about whether they really need access to private state to implement something.

---
In reply to: [166150283](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166150283) [](ancestors = 166150283,166090891,166073907)